concordium-consensus/tests/consensus/ConcordiumTests/KonsensusV1/TreeStateTest.hs

```diff
@@ -0,0 +1,908 @@
+            { thSender = dummyAccountAddress,
+              thPayloadSize = payloadSize payload,
+              thNonce = n,
+              thExpiry = 500,
+              thEnergyAmount = 5_000_000
+            }
+
+dummyTransaction :: Nonce -> Transaction
+dummyTransaction n = dummyTransactionWithPayload n (EncodedPayload "01234567890123456789")
+
+dummyTransactionBI :: Nonce -> BlockItem
+dummyTransactionBI = normalTransaction . dummyTransaction
+
+dummySuccessTransactionResult :: Nonce -> TVer.VerificationResult
+dummySuccessTransactionResult n =
+    TVer.Ok
+        TVer.NormalTransactionSuccess
+            { keysHash = Hash.hash "keys",
+              nonce = n
+            }
+
+dummySuccessCredentialDeployment :: TVer.VerificationResult
+dummySuccessCredentialDeployment = TVer.Ok TVer.CredentialDeploymentSuccess
+
+dummyRawUpdateInstruction :: UpdateSequenceNumber -> RawUpdateInstruction
+dummyRawUpdateInstruction usn =
+    RawUpdateInstruction
+        { ruiSeqNumber = usn,
+          ruiEffectiveTime = 0,
+          ruiTimeout = 0,
+          ruiPayload = MicroGTUPerEuroUpdatePayload 2
+        }
+
+dummyUpdateInstruction :: UpdateSequenceNumber -> UpdateInstruction
+dummyUpdateInstruction usn =
+    makeUpdateInstruction
+        (dummyRawUpdateInstruction usn)
+        (Map.fromList [(0, dummySigSchemeKeys)])
+
+dummyUpdateInstructionWM :: UpdateSequenceNumber -> WithMetadata UpdateInstruction
+dummyUpdateInstructionWM usn =
+    addMetadata ChainUpdate 0 $ dummyUpdateInstruction usn
+
+dummyChainUpdate :: UpdateSequenceNumber -> BlockItem
+dummyChainUpdate usn = chainUpdate $ dummyUpdateInstructionWM usn
+
+-- |A valid 'AccountCreation' with expiry 1596409020
+dummyAccountCreation :: AccountCreation
+dummyAccountCreation = readAccountCreation . BSL.fromStrict $ $(makeRelativeToProject "testdata/transactionverification/verifiable-credential.json" >>= embedFile)
+
+credentialDeploymentWM :: WithMetadata AccountCreation
+credentialDeploymentWM = addMetadata CredentialDeployment 0 dummyAccountCreation
+
+dummyCredentialDeployment :: BlockItem
+dummyCredentialDeployment = credentialDeployment credentialDeploymentWM
+
+testLookupLiveTransaction :: Spec
+testLookupLiveTransaction = describe "lookupLiveTransaction" $ do
+    it "present" $ do
+        assertEqual
+            "status transaction 1"
+            (Just $ Received 0 (dummySuccessTransactionResult 1))
+            $ lookupLiveTransaction (th 1) sd
+        assertEqual
+            "status transaction 2"
+            (Just $ Received 0 (dummySuccessTransactionResult 2))
+            $ lookupLiveTransaction (th 2) sd
+        assertEqual
+            "status transaction 3"
+            (Just $ Received 0 (dummySuccessTransactionResult 3))
+            $ lookupLiveTransaction (th 3) sd
+    it "absent"
+        $ assertEqual
+            "status transaction 4"
+            Nothing
+        $ lookupLiveTransaction (th 4) sd
+  where
+    th n = getHash (dummyTransactionBI n)
```

Comment:
Please avoid abbreviations

Reply:
transaction hash, nonce. I think it's ok?

Reply:
At the risk of being overly pedantic, it would really not cost you much to have a longer name here. It was clearly not OK since he took the trouble of pointing it out. In the time it took you to respond here you could have changed the name.

Especially since the methods here are generic so the types are unclear.